src/Modules/reservationPopUp.js

```diff
@@ -0,0 +1,179 @@
+const getReservation = async (itemId) => {
+  try {
+    const response = await fetch(`https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/zRod3rPxBRjxEDaYzujw/reservations?item_id=${itemId}`);
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    throw new Error(error);
+  }
+};
+
+const addReservation = async (itemId, name, dateStart, dateEnd) => {
+  const response = await fetch('https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/zRod3rPxBRjxEDaYzujw/reservations', {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({
+      item_id: itemId,
+      username: name,
+      date_start: dateStart,
+      date_end: dateEnd,
+    }),
+  });
+
+  if (!response.ok) {
+    const errorMessage = await response.text();
+    throw new Error(errorMessage);
+  }
+};
+
+const reservationPopUp = () => {
+  const reservBtn = Array.from(document.getElementsByClassName('btn-2'));
+  const pop = document.querySelector('.reserv-pop');
+  pop.setAttribute('style', 'display: none;');
+
+  const createPopup = async (data) => {
+    const popup = document.createElement('div');
+    popup.classList.add('popup');
+
+    const closeBtn = document.createElement('div');
+    closeBtn.classList.add('close-btn', 'text-4xl', 'py-2', 'cursor-pointer', 'text-right');
+    closeBtn.innerHTML = '&times;';
+    closeBtn.addEventListener('click', () => {
+      popup.remove();
+      pop.setAttribute('style', 'display: none;');
+    });
+
+    const popupImgContainer = document.createElement('div');
+    popupImgContainer.classList.add('img-container');
+
+    const popupImg = document.createElement('img');
+    popupImg.src = data.images?.coverart || data.images?.background || `https://dummyimage.com/400x400/000/1aff00&text=${data.title}`;
+
+    const popupTitle = document.createElement('h3');
+    popupTitle.classList.add('text-3xl', 'font-bold');
+    popupTitle.innerHTML = data.title;
+
+    const commentsTitle = document.createElement('h3');
+    commentsTitle.id = 'reservation-counter';
+
+    const reservations = getReservation(data.key).then((res) => {
+      if (res.error) {
+        commentsTitle.textContent = 'Reservations (0)';
+      } else {
+        commentsTitle.textContent = `Reservations (${res.length})`;
+        // console.log('---++++', res)
+        let htmlToAppend = '';
+        res.forEach((item) => {
+          const html = `<li>${item.date_start} - ${item.date_end} || by <strong> ${item.username}</strong></li>`;
+          htmlToAppend += html;
+        });
+        const reserveList = document.querySelector('.comment-list');
+        reserveList.innerHTML = htmlToAppend;
+      }
+    }).catch((err) => {
+      commentsTitle.textContent = 'Reservations (0)';
+      throw new Error(err);
+    });
+    commentsTitle.textContent = `Reservations (${reservations === undefined ? '0' : reservations?.length})`;
```

Comment:
- [ ] Great work, great design. However, there's littler bug here which I want you guys to look into and fix. I got `undefined` as the value of `Reservations (undefined)` when clicking on the reserve button after entered all the required values in the inputs boxes. The actual number of reservations for the particular item later show up when you reopen the item. Take a look at the screen recording for better understanding of what I am saying.

### App behaviour
[app-behave.webm](https://user-images.githubusercontent.com/26597432/235234746-e7cf3e0d-d02e-49fa-a18f-f0fd431a726b.webm)